### ISSUE TITLE
Basic templating feature for shared content

### ISSUE THREAD

**Issue by tauren** [gsantner/markor#1929]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **enhancement/feature request**. Not a [bug or question](https://github.com/gsantner/markor/issues/new/choose).
- [X] The topic is **not** already reported at [Issues](https://github.com/gsantner/markor/issues?q=#js-issues-search). _(I've searched it)_.
- [X] Markor **is** up to date. See [Releases](https://github.com/gsantner/markor/tags) for the latest version. Updates are available from [F-Droid](https://f-droid.org/en/packages/net.gsantner.markor/) and GitHub.
- [X] The wanted feature/enhancement is not present in the latest development version (git master). (Please [download](https://nightly.link/gsantner/markor/workflows/build-android-project/master) and try the test version of Markor, named **Mordor**. Don't worry; Markor and Mordor appear as completely separate applications. You can install both side-by-side, and Markor's settings are not touched. If your desired feature is present, you don't need to open this issue. The change will be part of the next Markor update.)

## Basic templating feature for shared content 

I add links and content to my QuickNote file all the time by using the system "Share -> Markor" feature. It would be really handy and save time if the content to share could automatically have extra metadata and markdown/formatting.

I personally would like all of my saved links to be saved into a bulleted list with the date and time I saved them. I don't want the horizontal rule between each. I can manually add date information, but it is tedious. However I can't remove the HR. I also add tags.

Here is an example "Share Template":

```
- {{date}} {{time}} {{CONTENT}}
```

Here's a bit more elaborate one:

```
- {{CONTENT}}
  - {{date}} {{time}}
```

This would append the following to the bottom of whatever file is selected (I include entering example tags here):

```
- [Some cool article](https://example.com/cool-article)
  - 2022-12-03 21:32 #article #cool #followup
```

Note that the template should REPLACE the `---` horizontal rule and linebreak that are currently added to the bottom of the file with the content. The default template would contain those things so the app continues to work the same way until the user changes the template. Default template might be:

```
----
{{CONTENT}}

```

It would be really nice to support common template tags, such as the ones Obsidian uses:
https://help.obsidian.md/Plugins/Templates

For instance, `{{date:MM-DD-YYYY}}`. It supports all of the Moment.js format strings.

This could be expanded to include other templating commands, but I recognize the desire to keep Markor simple.

### Information

Android version: 11
Device: Pixel 3
App Version: 2.10.6

### Source

Google Play

### Format / File type

Markdown

### Additional info / Log

```shell
-
```

**Comment by harshad1** [comment]
Settings -> other -> share into format. It supports date interpolation and text positioning.

**Comment by tauren** [comment]
Oh sweet, I guess I missed it in the settings! Thanks for pointing it out.

Is there more detailed documentation on this somewhere? It's a bit confusing and finicky, and doesn't appear to follow the text description in the Share into settings screen regarding `'%s'`. That screen doesn't explain that `{{text}}` doesn't work if you also use a date format.

For instance, these don't render correctly:

`'/n-' yyyy-MM-dd '%s'`
`'/n-' yyyy-MM-dd {{text}}`

But I was able to get this template to work (note there is a space at the end, after which the link is appended):

`'/n-' yyyy-MM-dd `

**Comment by harshad1** [comment]
There may be bugs in the implementation. I plan to look into this soon

**Comment by harshad1** [comment]
Fyi, there was a bug. It's fixed in #1937. It'll take a little while before this gets merged and pushed out into a release.

**Comment by gsantner** [comment]
Closing, there are some possibilities now.
